Download reports UI - dynamic year dropdown added

seedyjahateh opened the pull request:
We created a reports UI under the Admin tab. UI includes a download button and an input bar for the academic year of the report you want to download.
The year options in the dropdown are taken from the database.

Fixes #1104

andersoncedu committed:
Removed spreadsheet ui from this branch.

andersoncedu committed:
Removed stats.py

WackyWeaver committed:
Pulled development

seedyjahateh committed:
All done except the path extraction from config. Still in progress.

seedyjahateh committed:
We created a reports UI under Admin tab. UI includes a download button and an input bar for the year to download the report.

seedyjahateh committed:
Merge branch 'development' into download-reports-UI

BrianRamsay commented on app/controllers/admin/routes.py:
```diff
@@ -38,6 +38,25 @@
 
 from app.controllers.admin import admin_bp
+from flask import send_file, render_template, request
+from app.controllers.admin import admin_bp
```
Remove redundant imports

seedyjahateh replied:
removed

BrianRamsay commented on app/controllers/admin/routes.py:
```diff
@@ -38,6 +38,25 @@
+def reports():
+    academic_years = Term.select(Term.academicYear).distinct().order_by(Term.academicYear.desc())
+    academic_years = list(map(lambda t: t.academicYear, academic_years))
+    return render_template("/admin/reports.html", academic_years=academic_years)
```
Our variables are camelCase in our application

seedyjahateh replied:
Changed names to camelCase.

BrianRamsay commented on app/controllers/admin/routes.py:
```diff
@@ -38,6 +38,25 @@
+    return render_template("/admin/reports.html", academic_years=academic_years)
+
+@admin_bp.route('/download')
+def download_file():
```
Same with function names

seedyjahateh replied:
Done

BrianRamsay commented on app/controllers/admin/routes.py:
```diff
@@ -38,6 +38,25 @@
+    academic_years = list(map(lambda t: t.academicYear, academic_years))
+    return render_template("/admin/reports.html", academic_years=academic_years)
+
+@admin_bp.route('/download')
```
Route should look related to the page that got us here (`/admin/reports`)

seedyjahateh replied:
How about '/admin/reports/download'?

BrianRamsay commented on app/controllers/admin/routes.py:
```diff
@@ -38,6 +38,25 @@
+def download_file():
+    academic_year = request.args.get('academic_year', '2023-2024')  # Default academic year if not provided
+    filepath = createSpreadsheet(academic_year)
+    filepath = os.path.abspath(filepath)
```
Let's combine this into one line since it fits

seedyjahateh replied:
Done.

BrianRamsay commented on app/templates/admin/reports.html:
```diff
@@ -0,0 +1,31 @@
+          text-align: center;
+          margin-top: 50px;
+      }
+  </style>
```
This needs to be in a css file.

seedyjahateh replied:
Used built-in bootstrap

BrianRamsay commented on app/templates/admin/reports.html:
```diff
@@ -0,0 +1,31 @@
+                {% endfor %}
+            </select>
+        </div>
+        <button type="submit" class="btn btn-primary" onclick="confirmDownload()">Download Spreadsheet</button>
```
Let's say "Download Annual Report"

BrianRamsay replied:
Maybe with an excel icon in the button

seedyjahateh replied:
Changed

seedyjahateh committed:
Made changes requested in the PR conversations.

seedyjahateh committed:
Merge branch 'download-reports-UI' of https://github.com/BCStudentSoftwareDevTeam/celts into download-reports-UI

BrianRamsay reviewed (changes requested):
The select box doesn't need to be full screen width. Let's use our form-control styles for the label and select box. 

Download button can just say Download, with the icon in front of it.

seedyjahateh committed:
Reduced the size of 'select box', used built-in bootstrap for styling the 'select box' and 'label', and renamed the download button to 'Download' and moved the Excel icon infront of the name.

seedyjahateh commented:
> The select box doesn't need to be full screen width. Let's use our form-control styles for the label and select box.
> 
> Download button can just say Download, with the icon in front of it.

Reduced the size of 'select box', used built-in bootstrap for styling the 'select box' and 'label', and renamed the download button to 'Download' and moved the Excel icon infront of the name.

MImran2002 commented:
suggestion: 
![image](https://github.com/BCStudentSoftwareDevTeam/celts/assets/142051335/578b13c1-0ee1-4ecd-a9ee-7ed218cd7cf5)
I would recommend as this is a dropdown UI to have dropdown design. The code is perfect it's just I feel a design perspective 
link: https://getbootstrap.com/docs/4.0/components/dropdowns/

kudos that it all works very well. the download button works too!

WackyWeaver committed:
changed from form-control to form-select

BrianRamsay reviewed (changes requested):
Looks good! However, the academic year is not being used properly when the spreadsheet is generated

WackyWeaver committed:
fixed the post/get method issue

WackyWeaver commented:
> Looks good! However, the academic year is not being used properly when the spreadsheet is generated

We found that this needs to be a whole other issue with spreadsheets.py

BrianRamsay committed:
Merge branch 'development' into download-reports-UI
